src/beacon/app/data_transmission_endpoint.py

```
@@ -1,499 +1,732 @@
-                hs.hour,
-                COALESCE(hr.reading_count, 0) AS actual_readings,
-                COALESCE(hr.active_device_count, 0) AS transmitting_devices,
-                (SELECT count FROM active_devices) AS total_devices,
-                (SELECT count FROM active_devices) * 1 AS expected_readings_per_hour
-            FROM hour_series hs
-            LEFT JOIN hourly_readings hr ON hs.hour = hr.hour
-            ORDER BY hs.hour
-        """)
+    except Exception as e:
+        print(f"Error in comprehensive transmission analytics: {str(e)}")
+        raise HTTPException(status_code=500, detail=f"Failed to fetch comprehensive analytics: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)